client/src/upload/Upload.tsx

```diff
@@ -0,0 +1,66 @@
+import axios from "axios";
+import { useState } from "react";
+import { Link } from "react-router-dom";
+
+// Upload a file, send it to backend, and display the response
+export default function Upload() {
+    const [file, setFile] = useState<File | null>(null);
+    const [response, setResponse] = useState<string | null>(null);
+    const [error, setError] = useState<string | null>(null);
+
+    // send get request to /test on port 3001
+    const handleClick = () => {
+        const reqBody = {
+            file: file,
+            "test": "test123"
+        };
+
+        axios.get("http://127.0.0.1:3001/pdf", { data: reqBody })
```

Comment:
Then for all the requests everywhere in the project, do `axios.get('/pdf')` etc would be sufficient. Base url will change accordingly under different environment.